opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -140,13 +140,20 @@ impl LoggerProvider {
 struct LoggerProviderInner {
     processors: Vec<Box<dyn LogProcessor>>,
     resource: Resource,
+    is_shutdown: AtomicBool,
 }
 
 impl Drop for LoggerProviderInner {
     fn drop(&mut self) {
-        for processor in &mut self.processors {
-            if let Err(err) = processor.shutdown() {
-                global::handle_error(err);
+        if self
```

Comment:
i don't think drop should modify the is_shutdown flag here. it should just call the shutdown() method if not already shutdown, and nothing more. the shutdown() method imlementation should know how to handle shutdown() and invoke processors etc - that logic is not required to be duplicated in drop.

Reply:
shutdown() is part of LoggerProvider, and can't be called from within LoggerProviderInner.  We can instead create a private `shutdown()` method inside the LoggerProviderInner, and let it be called from both LoggerProvider::shutdown() and LoggerProviderInner::drop().

Reply:
i think we should move shutdown() logic to be inside inner one, with the loggerprovider delgating everything, including shutdown to the inner.

LoggerProvier/Meterprovider/Tracerprovider - they are just shared ref to the actual provider. So entire logic should be moved inside inner. When working on periodicreader, I think i managed to achieve the same for PeriodicReader.

![image](https://github.com/user-attachments/assets/facda6cc-9fd3-42b3-afef-0a152bf00008)

If this makes sense, then lets do it everywhere, starting with LoggerProvider

Reply:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/metrics/meter_provider.rs#L25 The doc comment here is something we should replicate too.

Reply:
Have added a crate private method in LoggerProviderInner, and let it be called from both Drop and Shutdown. The error/warn/debug handling can be ignored for now (didn't change it), will handle it in the separate PR.